app/graphql/albums/middleware.js

```diff
@@ -0,0 +1,13 @@
+const { checkAuthenticated } = require('../../helpers');
+
+const buyAlbum = (resolve, root, args, context) => {
+  const { user } = context;
+  checkAuthenticated(user);
+  return resolve(root, args, { user });
```

Comment:
Be careful here. Imagine you save more information in the context, besides `user`, in that case this middleware will omit that information. Return the entire context please